File: storm-core/src/test/java/st/orm/core/model/VetSpecialty.java
package st.orm.core.model;

import static st.orm.GenerationStrategy.NONE;

import lombok.Builder;
import st.orm.Entity;
import st.orm.FK;
import st.orm.PK;
import st.orm.Persist;

@Builder(toBuilder = true)
public record VetSpecialty(
        @PK(generation = NONE) VetSpecialtyPK id,  // Implicitly @Inlined
        @Persist(insertable = false) @FK Vet vet,
        @Persist(insertable = false) @FK Specialty specialty) implements Entity<VetSpecialtyPK> {
    public VetSpecialty(VetSpecialtyPK pk) {
        //noinspection DataFlowIssue
        this(pk, null, null);
    }
}

File: storm-kotlinx-serialization/src/test/kotlin/st/orm/serialization/model/VetSpecialty.kt
package st.orm.serialization.model

import kotlinx.serialization.Serializable
import st.orm.*

@Serializable
internal data class VetSpecialty(
    @PK(generation = GenerationStrategy.NONE) val id: VetSpecialtyPK, // Implicitly @Inlined
    @FK @Persist(insertable = false) val vet: Vet,
    @FK @Persist(insertable = false) val specialty: Specialty,
) : Entity<VetSpecialtyPK>
